[2023-04-27T08:32:24.732+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark-example.sql_job scheduled__2023-04-26T00:00:00+00:00 [queued]>
[2023-04-27T08:32:24.740+0000] {taskinstance.py:1090} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark-example.sql_job scheduled__2023-04-26T00:00:00+00:00 [queued]>
[2023-04-27T08:32:24.740+0000] {taskinstance.py:1288} INFO - 
--------------------------------------------------------------------------------
[2023-04-27T08:32:24.740+0000] {taskinstance.py:1289} INFO - Starting attempt 3 of 3
[2023-04-27T08:32:24.740+0000] {taskinstance.py:1290} INFO - 
--------------------------------------------------------------------------------
[2023-04-27T08:32:24.754+0000] {taskinstance.py:1309} INFO - Executing <Task(SparkSqlOperator): sql_job> on 2023-04-26 00:00:00+00:00
[2023-04-27T08:32:24.760+0000] {standard_task_runner.py:55} INFO - Started process 304 to run task
[2023-04-27T08:32:24.764+0000] {standard_task_runner.py:82} INFO - Running: ['***', 'tasks', 'run', 'spark-example', 'sql_job', 'scheduled__2023-04-26T00:00:00+00:00', '--job-id', '25', '--raw', '--subdir', 'DAGS_FOLDER/spark-example.py', '--cfg-path', '/tmp/tmpt43542ns']
[2023-04-27T08:32:24.766+0000] {standard_task_runner.py:83} INFO - Job 25: Subtask sql_job
[2023-04-27T08:32:24.822+0000] {task_command.py:389} INFO - Running <TaskInstance: spark-example.sql_job scheduled__2023-04-26T00:00:00+00:00 [running]> on host 652c4a0239ad
[2023-04-27T08:32:24.886+0000] {taskinstance.py:1518} INFO - Exporting the following env vars:
AIRFLOW_CTX_DAG_OWNER=***
AIRFLOW_CTX_DAG_ID=spark-example
AIRFLOW_CTX_TASK_ID=sql_job
AIRFLOW_CTX_EXECUTION_DATE=2023-04-26T00:00:00+00:00
AIRFLOW_CTX_TRY_NUMBER=3
AIRFLOW_CTX_DAG_RUN_ID=scheduled__2023-04-26T00:00:00+00:00
[2023-04-27T08:32:25.021+0000] {spark_sql.py:177} INFO - /home/***/.local/lib/python3.7/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found

[2023-04-27T08:32:25.022+0000] {spark_sql.py:177} INFO - JAVA_HOME is not set

[2023-04-27T08:32:25.036+0000] {taskinstance.py:1776} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/operators/spark_sql.py", line 97, in execute
    self._hook.run_query()
  File "/home/airflow/.local/lib/python3.7/site-packages/airflow/providers/apache/spark/hooks/spark_sql.py", line 183, in run_query
    f"Cannot execute '{self._sql}' on {self._master} (additional parameters: '{cmd}'). "
airflow.exceptions.AirflowException: Cannot execute 'select 1+1' on local (additional parameters: ''). Process exit code: 1.
[2023-04-27T08:32:25.039+0000] {taskinstance.py:1332} INFO - Marking task as FAILED. dag_id=spark-example, task_id=sql_job, execution_date=20230426T000000, start_date=20230427T083224, end_date=20230427T083225
[2023-04-27T08:32:25.053+0000] {standard_task_runner.py:105} ERROR - Failed to execute job 25 for task sql_job (Cannot execute 'select 1+1' on local (additional parameters: ''). Process exit code: 1.; 304)
[2023-04-27T08:32:25.096+0000] {local_task_job.py:212} INFO - Task exited with return code 1
[2023-04-27T08:32:25.114+0000] {taskinstance.py:2596} INFO - 0 downstream tasks scheduled from follow-on schedule check
